clippy_lints/src/redundant_pattern_matching.rs

```diff
@@ -86,19 +85,20 @@ fn find_sugg_for_if_let<'a, 'tcx>(
         _ => return,
     };
 
-    let maybe_semi = if has_else { "" } else { ";" };
-
     span_lint_and_then(
         cx,
         REDUNDANT_PATTERN_MATCHING,
         arms[0].pat.span,
         &format!("redundant pattern matching, consider using `{}`", good_method),
         |diag| {
-            let span = expr.span.to(op.span);
+            // in the case of WhileLetDesugar expr.span == op.span incorrectly.
+            // this is a workaround to restore true value of expr.span
+            let expr_span = expr.span.to(arms[1].span);
+            let span = expr_span.until(op.span.shrink_to_hi());
             diag.span_suggestion(
                 span,
                 "try this",
-                format!("{}.{}{}", snippet(cx, op.span, "_"), good_method, maybe_semi),
+                format!("{} {}.{}", keyword, snippet(cx, op.span, "_"), good_method),
                 Applicability::MaybeIncorrect, // snippet
```

Comment:
I think this is
```suggestion
                Applicability::MachineApplicable, // snippet
```
now

Reply:
Agree. Fixed.